Support MFC-J430W, fix many bugs, smarter page post-processing

rumpeltux opened the pull request:
1) Submodules not initialized (#5):
$ make
make: *** No rule to make target 'ber/ber.c', needed by 'all'.  Stop.

2)
connection.c: In function ‘brother_conn_poll’:
connection.c:179:25: error: ISO C forbids empty initializer braces [-Werror=pedantic]
     struct pollfd pfd = {};

3)
device_handler.c: In function ‘device_handler_add_device’:
device_handler.c:180:9: error: ‘dev’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
         free(dev);
         ^~~~~~~~~

rumpeltux committed:
Fix compilation errors:

1) Submodules not initialized (#5):
$ make
make: *** No rule to make target 'ber/ber.c', needed by 'all'.  Stop.

2)
connection.c: In function ‘brother_conn_poll’:
connection.c:179:25: error: ISO C forbids empty initializer braces [-Werror=pedantic]
     struct pollfd pfd = {};

3)
device_handler.c: In function ‘device_handler_add_device’:
device_handler.c:180:9: error: ‘dev’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
         free(dev);
         ^~~~~~~~~

rumpeltux committed:
Support MFC-J430W

rumpeltux committed:
Fix includes for compilation.

rumpeltux committed:
Import

rumpeltux committed:
Local fixes

rumpeltux committed:
Fix tests

rumpeltux committed:
Default loglevel to INFO. Add a debug flag (-d). Print hexdumps to stderr.

rumpeltux committed:
Fix test

rumpeltux committed:
Catch errors in device_handler_extract_string

rumpeltux committed:
Move unittest config to testdata/

rumpeltux committed:
Add RLE test document.

rumpeltux committed:
Fix snmp.c linkage bug

rumpeltux committed:
Init datachannel params when itemconfig is set

rumpeltux committed:
Fix LOG format send/recv

rumpeltux committed:
Exit scan script immediately on errors. Remove temp files on success.

rumpeltux committed:
Fix timeout logic.

rumpeltux committed:
Misc fixes. Reset page_id before scan, don't wait for final userhook script.

rumpeltux committed:
Limit the number of registration messages sent per snmp packet.

rumpeltux committed:
Processing script fixes

rumpeltux committed:
Fix compilation

rumpeltux committed:
Add an example script for remote processing.

rumpeltux committed:
Allow spaces in scan.func arguments.

rumpeltux committed:
Add a scan-id to group pages of a specific scan.

rumpeltux committed:
Don't use a new appnum for each registration.

rumpeltux committed:
Add some more debug logging.

rumpeltux committed:
Ignore low-on-ink errors.

rumpeltux committed:
Update ocr-fake-duplex-adf docs.

rumpeltux committed:
Make test a phony target

rumpeltux committed:
Fix clang compilation warnings (make CC=clang CFLAGS="-Wall")

rumpeltux committed:
Add iwyu support

rumpeltux committed:
make fix_include

rumpeltux committed:
Add --nosafe_headers to iwyu

rumpeltux committed:
Add 'make install' target

rumpeltux committed:
Fix compilation for arm

rumpeltux committed:
Don't modify /etc config if files exist

rumpeltux committed:
Ignore empty ink message.

rumpeltux committed:
Ignore empty ink message.

rumpeltux committed:
Ignore empty ink message.

chickenandpork commented:
@darsto in my biased opinion, one of the best things about an open source driver is that contributions can make it better and can get everyone slowly merged to cooperatively move forward.

I hope PRs can be merged.  I ran into exactly the errors mentioned in this PR.  Is there anything needed to merge this ?

(Thanks, @rumpletux BTW)

sgtell committed:
support variant status word 40000 returned by HL-L2380DW
after observing that the brother HL-L2380DW returns 40000 (0x9c40) in response
to a status query, accepting that as a valid response lets brother-scand
work nicely with this model as well.
Documented HL-L2380DW as working in Readme.MD

DestyNova committed:
README: Switch to new repo URL (#2)

rumpeltux committed:
Add UFW profile.

rumpeltux committed:
Merge remote-tracking branch 'github/master'

Commit:
fix compiler error

Commit:
Treat ECONNRESET as connection lost too and reconnect.

Commit:
Makefile: change USER to PRINTUSER to avoid confusion with the regular hagen env.

Commit:
Note that clang is a prerequisite for running tests.
